packages/web3-eth-contract/types/index.d.ts

```diff
@@ -29,6 +29,8 @@ export class Contract {
         options?: ContractOptions
     );
 
+    static setProvider(provider: string): void;
```

Comment:
The `setProvider` accepts an additional parameter for accounts as well. 

https://github.com/ChainSafe/web3.js/blob/832271a38e555fa62c843a05aba4eb5514f53d8a/packages/web3-eth-contract/src/index.js#L338

Reply:
Hiya, thanks good point. Added TS defs for optional `acconts` parameter as suggested.

Reply:
And if we're going to do this properly then parameter `provider` is of type `provider` too (from `web3-core')